packages/iframe-plugin/src/useHtmlIframeProps.ts

```diff
@@ -80,7 +80,12 @@ export default function useHtmlIframeProps(
 
   const source = htmlAttribs.srcdoc
     ? { html: htmlAttribs.srcdoc as string, baseUrl: documentBaseUrl }
-    : { uri: normalizedUrl };
+    : {
+        headers: {
+          ...resolvedConfig.webViewProps?.source?.headers
```

Comment:
I'd rather have an function which takes a source `uri` as parameter, and returns optional headers, because the logic might depend on domains. Because the logic could be generalized to images and other media, I think I'm going to implement a function upstream in RNRH, something like : 

```ts
type EmbeddedHeadersProvider = (tagName: string, embeddedUri: string) => Headers | null | void;
```
And then we'll be able to consume this function from `sharedProps`. I'll keep you updated when that lands.

Reply:
You're right. This kind of function would be great to keep a generic approach that would suits to different uses and domains.

Reply:
Commit is ready,  still need to work on some fixes for v6.1.0 and we'll be good https://github.com/meliorence/react-native-render-html/commit/2ea1ca6a0fc58aca59ff4fbc080dfeb7f739ff88